### PR TITLE
[NVPTX][NFC] cleanup dead vars, use MAKE_CASE

### PR DESCRIPTION
**Pull request by AlexMaclean**
Cleanup some dead variables. In addition, switch to a `MAKE_CASE` macro, similar to other targets, to reduce boilerplate.

### CODE REVIEW

**Comment by AlexMaclean** [comment]
@Artem-B, could you please review this when you have a moment. Thanks!

File: llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp [hunk 1]
```diff
@@ -858,623 +858,433 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
 }
 
 const char *NVPTXTargetLowering::getTargetNodeName(unsigned Opcode) const {
+
+#define MAKE_CASE(V)                                                           \
```
Comment (Artem-B) [inline llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp:862]: I wonder if we could do it the way WebAssembly does: https://github.com/llvm/llvm-project/blob/ab4a793e8bc78f50f9f104c9c732e2dd91bf70a2/llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp#L719

This approach is fine, but it's still too much of a boilerplate.

File: llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp [hunk 2]
```diff
@@ -858,623 +858,433 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
...
+    MAKE_CASE(NVPTXISD::Suld3DV2I32Zero)
+    MAKE_CASE(NVPTXISD::Suld3DV2I64Zero)
+    MAKE_CASE(NVPTXISD::Suld3DV4I8Zero)
+    MAKE_CASE(NVPTXISD::Suld3DV4I16Zero)
+    MAKE_CASE(NVPTXISD::Suld3DV4I32Zero)
   }
   return nullptr;
+
+#undef MAKE_CASE
+  return nullptr;
```
Comment (Artem-B) [inline llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp:1287]: duplicate return.

**Review by Artem-B — COMMENTED** [review]
Thank you for the clean-up. Few nits, but looks good otherwise.

File: llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp (hunk 2 above)
Comment (AlexMaclean) [reply llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp]: Fixed.

File: llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp (hunk 1 above)
Comment (AlexMaclean) [reply llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp:862]: Interesting. Something to consider for sure.

**Comment by AlexMaclean** [comment]
@Artem-B, could you please merge this on my behalf? Thanks!

**Review by Artem-B — APPROVED** [review]

**Pull request merged**